### DISCUSSION TITLE
dlt_register_app return value does not reflect `/tmp/dlt cannot be opened` status

### DISCUSSION THREAD

**Discussion by jwinarske** [COVESA/dlt-daemon#665] [General]
When calling `dlt_register_app` it returns `Ok` and prints out `~FIFO /tmp/dlt cannot be opened. Retrying later...`

Why is this not returned as a unique response value from `dlt_register_app`?

Reasoning I think it should:
1. I want to change program flow based on this state
2. I want to manage what is printed to cout/cerr

**Comment by minminlittleshrimp** [comment]
Hello @jwinarske 
TMHO, I think it is because DLT following Autosar specs, or at least that's what I believe. The point is libdlt has it own buffer, and when no daemon available, any applications trying to log will be registered successfully and start logging to libdlt buffer. This makes sure when daemon done 2 phases of setting up, all registered apps in list can be forwarded immediately, and also logs in libdlt will be fetched to daemon. Thus, this is useful and can say, be a must-have feature in logging, esp. when daemon not yet start in bootup phase but all other apps/services start logging.
In conclusion, I believe this is the app registration mechanism from DLT Daemon.
Regards
